exampleSite/content/posts/carousel.md

```diff
@@ -0,0 +1,44 @@
++++
+title = "Carousel"
+description = "Post with carousel."
+summary = ""
+toc = true
+authors = ["Example Author"]
+tags = ["image"]
+categories = ["themes"]
+series = ["Themes Guide"]
+date =  "2020-05-26"
+lastmod = "2020-05-26"
+draft = false
+
+[carousel]
+  enable = 1
+  images = [
+	{ weight = 1, image = "../carousel1.png", description = "Caption of the first image." },
+	{ weight = 2, image = "../carousel2.png", description = "Caption of the second image." },
+	{ weight = 3, image = "../carousel3.png", description = "The next image does not have a caption." },
+	{ weight = 4, image = "../carousel4.png" }
+  ]
```

Comment:
What do you think about the relationship between featuredImage and carousel? I am thinking whether there is a use case where these two attributes are used at the same time? If not, we can merge it into one attribute.

Btw, the `enable` is not necessary because you can use `{{ if .Params.carousel }}` or `{{ with .Params.carousel }}` to judge.

And I think this can be refactorred as `carousel = ["image_url_1", "image_url_2"]`. The order is weight. and for others who are willing to add a description, `carousel = [{url= "carousel1.jpg", description = "Caption of the first image." }, ...]` can be used. And use functions like `reflect.IsMap` in template to determine whether the item in the list is a string or a map.

Reply:
Agreed. I assume something like this was in your mind: fabf93a